datascrubber/cli.py

```diff
@@ -0,0 +1,300 @@
+        '--share-with',
+        required=False,
+        type=str,
+        nargs='+',
+        help="AWS account IDs to share scrubbed snapshots with",
+    )
+
+    return parser.parse_args()
+
+
+def configure_logging(mode, level_name):
+    level = {
+        'CRITICAL': logging.CRITICAL,
+        'ERROR': logging.ERROR,
+        'WARNING': logging.WARNING,
+        'INFO': logging.INFO,
+        'DEBUG': logging.DEBUG
+    }.get(level_name)
+
+    def log_config_console():
+        logging.basicConfig(
+            format='%(asctime)s | %(levelname)s | %(threadName)s | %(name)s %(funcName)s | %(message)s',
+            level=level,
+        )
+        return
+
+    def log_config_syslog():
+        logger = logging.getLogger()
+        logger.setLevel(level)
+        handler = logging.handlers.SysLogHandler(address='/dev/log')
+        formatter = logging.Formatter('%(name)s %(funcName) | %(levelname)s | %(threadName)s | %(message)s')
+        handler.setFormatter(formatter)
+        logger.addHandler(handler)
+        return
+
+    if mode is not None:
+        if mode == 'syslog':
+            return log_config_syslog()
+        if mode == 'console':
+            return log_config_console()
+
+    if sys.stdout.isatty():
+        return log_config_console()
+
+    elif os.environ.get('LAMBDA_TASK_ROOT') is None:
+        return log_config_console()
+
+    else:
+        return log_config_syslog()
+
+
+def worker(dbms, hostname=None, instance=None, snapshot=None, target_accounts=[]):
+    logger = logging.getLogger()
+    logger.info("Spawned new worker thread")
+
+    # We need a boto3 session per thread
+    # https://boto3.readthedocs.io/en/latest/guide/resources.html#multithreading-multiprocessing
+    session = boto3.session.Session()
+    rds_client = session.client('rds')
+    workspace = None
+
+    try:
+        snapshot_finder = RdsSnapshotFinder(
+            dbms,
+            boto3_session=session,
+            hostname=hostname,
+            source_instance_identifier=instance,
+            snapshot_identifier=snapshot,
+        )
+
+        workspace = ScrubWorkspaceInstance(
+            snapshot_finder,
+            session,
+        )
+
+        if dbms == 'mysql':
+            task_manager = Mysql(workspace)
+        elif dbms == 'postgresql':
+            task_manager = Postgresql(workspace)
+        else:
+            raise Exception("DBMS not supported: %s" % dbms)
+
+        success = True
+        for task in task_manager.get_viable_tasks():
+            success = task_manager.run_task(task)
+            if not success:
+                logger.error(
+                    "Task %s failed. A final snapshot will not be generated, "
+                    "in case sensitive data remains.",
+                    task
+                )
+                break
+
+        workspace.cleanup(create_final_snapshot=success)
+        if success:
+            if target_accounts is not None and len(target_accounts) >= 1:
```

Comment:
Won't `len(target_accounts) >= 1` deal with the `target_accounts is not None` case too?

Reply:
Maybe, I'll check.. (`len(None)` might throw an error).

Reply:
Yup 😄 
```
>>> len(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: object of type 'NoneType' has no len()
```